libyul/AsmAnalysis.h

```diff
@@ -60,14 +60,16 @@ class AsmAnalyzer
 		AsmAnalysisInfo& _analysisInfo,
 		langutil::ErrorReporter& _errorReporter,
 		Dialect const& _dialect,
+		std::string const& _objectName,
 		ExternalIdentifierAccess::Resolver _resolver = ExternalIdentifierAccess::Resolver(),
 		std::set<std::string> _dataNames = {}
 	):
 		m_resolver(std::move(_resolver)),
 		m_info(_analysisInfo),
 		m_errorReporter(_errorReporter),
 		m_dialect(_dialect),
-		m_dataNames(std::move(_dataNames))
+		m_dataNames(std::move(_dataNames)),
+		m_objectName(_objectName)
```

Comment:
Since the name gets stored, it would be best to pass it by value rather than reference in all these functions/constructors and use `move`:

```suggestion
		m_objectName(std::move(_objectName))
```

Reply:
For that to make sense you also have to pass it by value (i.e. `std::string` rather than `std::string const&`).